src/encoded/static/components/navigation/components/SearchBar.js

```diff
@@ -28,71 +28,109 @@ export class SearchBar extends React.PureComponent{
         return (typedSearchQuery && typeof typedSearchQuery === 'string' && typedSearchQuery.length > 0) || false;
     }
 
-    constructor(props){
+    static isBrowseOrSearchPage(href) {
+        return href && typeof href === 'string' && (href.indexOf('/browse/') > -1 || href.indexOf('/search/') > -1);
+    }
+
+    constructor(props) {
         super(props);
-        this.toggleSearchAllItems   = this.toggleSearchAllItems.bind(this);
-        this.onSearchInputChange    = this.onSearchInputChange.bind(this);
-        this.onResetSearch          = this.onResetSearch.bind(this);
-        this.onSearchInputBlur      = this.onSearchInputBlur.bind(this);
+        this.toggleSearchAllItems = this.toggleSearchAllItems.bind(this);
+        this.onSearchInputChange = this.onSearchInputChange.bind(this);
+        this.onResetSearch = this.onResetSearch.bind(this);
+        this.onSearchInputBlur = this.onSearchInputBlur.bind(this);
         this.selectItemTypeDropdown = this.selectItemTypeDropdown.bind(this);
+        this.getDropdownTitleText = this.getDropdownTitleText.bind(this);
+        this.getActionType = this.getActionType.bind(this);
 
         var initialQuery = '';
-        if (props.href){
+        if (props.href) {
             initialQuery = searchFilters.searchQueryStringFromHref(props.href) || '';
         }
         this.state = {
-            'searchAllItems'    : props.href && navigate.isSearchHref(props.href),
-            'typedSearchQuery'  : initialQuery
+            'searchItemType': props.href && navigate.isSearchHref(props.href) ? 'all' : 'sets',
+            'typedSearchQuery': initialQuery,
         };
     }
 
-    componentDidUpdate(pastProps){
+    componentDidUpdate(pastProps) {
         const { href } = this.props;
-        if (pastProps.href !== href){
+        if (pastProps.href !== href) {
             const query = searchFilters.searchQueryStringFromHref(href) || '';
-            this.setState(function({ typedSearchQuery }){
-                if (query !== typedSearchQuery){
-                    return { 'typedSearchQuery' : query };
+            this.setState(function ({ typedSearchQuery }) {
+                if (query !== typedSearchQuery) {
+                    return { 'typedSearchQuery': query, 'searchItemType': 'sets' };
                 }
                 return null;
             });
         }
     }
 
-    toggleSearchAllItems(willSearchAllItems = !this.state.searchAllItems){
-        this.setState(function({ searchAllItems }){
-            if (typeof willSearchAllItems === 'boolean'){
-                if (willSearchAllItems === searchAllItems){
-                    return null;
+    getCurrentResultsSearchQuery(hrefParts) {
+        if (!hrefParts) {
+            hrefParts = url.parse(this.props.href, true);
+        }
+        return (hrefParts && hrefParts.query && hrefParts.query.q) || null;
+    }
+
+    getDropdownTitleText() {
+        const { href } = this.props;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['href' is assigned a value but never used. (no-unused-vars)](https://app.codacy.com/app/4dn/fourfront/pullRequest?prid=4513406)